src/core/lib/event_engine/cf_engine/cfstream_endpoint.cc

```diff
@@ -331,6 +331,10 @@ void CFStreamEndpointImpl::DoWrite(
   size_t total_written_size = 0;
   for (size_t i = 0; i < data->Count(); i++) {
     auto slice = data->RefSlice(i);
+    if (slice.size() == 0) {
+        continue;
+    }
+
     size_t written_size =
```

Comment:
@HannahShiSFB Could you recommend a fix here if it's very simple? If not we can always do a follow-up fix.

Reply:
It can the the same as the read path:
https://github.com/grpc/grpc/pull/37255/files/1d828d253190bfd4293623248e679d39500c7543#diff-e0c8c9a1eaa8807b6d1cd56b8fab35fa02b3ffcae4e4f422cbf56fa1c55b1cbeL294